.github/workflows/ci-test.yml

```diff
@@ -205,8 +205,16 @@ jobs:
            echo "/usr/local/opt/bison/bin" >> $GITHUB_PATH
 
     - name: Build (MacOS)
+      # for some reason, brew does not install protobuf in a way cmake would be
+      # able to find it. Also, the libarary needs to be specified manually in
+      # order for the compiler to be correctly linked with protobuf.
       run: |
-          ./bootstrap.sh -DENABLE_GC=ON -DCMAKE_BUILD_TYPE=RELEASE -DENABLE_UNIFIED_COMPILATION=ON && cd build && make -j2
+          PROTOBUF=`brew --prefix --installed protobuf`
+          ./bootstrap.sh -DENABLE_GC=ON -DCMAKE_BUILD_TYPE=RELEASE \
+              -DENABLE_UNIFIED_COMPILATION=ON \
+              -DProtobuf_INCLUDE_DIR=$PROTOBUF/include \
+              -DProtobuf_LIBRARY=$PROTOBUF/lib/libprotobuf.dylib
```

Comment:
For some reason, the cmake is not able to correctly link to the library even if given path to the right directory with libraries. This is the problem also present in #3895. I've used this trick which I have learned on Archlinux where the detection is also broken.

Reply:
https://stackoverflow.com/questions/34795816/hinting-findname-cmake-files-with-a-custom-directory

Did you try playing with cmake prefix path?

Reply:
This problem is caused by a recent change with in the protobuf formula (https://github.com/Homebrew/homebrew-core/pull/122277) and suspect that it might be fixed in the future.